### PR TITLE
RUBY-900 Don't add discovered servers to the cluster with Single topology

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

File: lib/mongo/cluster.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def ==(other)
     # @since 2.0.0
     def add(host)
       address = Address.new(host)
-      unless addresses.include?(address)
+      unless addresses.include?(address) || (@topology.single? && address.seed != @topology.seed)
```
Comment (estolfo) [inline lib/mongo/cluster.rb]: @durran This line can probably be improved, let me know if you have any other ideas.

**Comment by jeromegn** [comment]
I believe that would fix it!

**Comment by estolfo** [comment]
cool, thanks so much for helping out @jeromegn !

File: lib/mongo/cluster.rb (hunk 1 above)
Comment (durran) [inline lib/mongo/cluster.rb]: I would probable change the logic to to make it a bit less confusing as well as extracting a method on the second part...

```ruby
if !addresses.include?(address) || direct_connection?(address)
  #...
end
```

Comment (estolfo) [inline lib/mongo/cluster.rb]: done, thanks

**Pull request merged**